content/en/docs/concepts/logic.md

```diff
@@ -492,30 +515,36 @@ it is guaranteed that this will be the final result.
 If a value is not concrete (like `string`), it is clear the search
 for a concrete value is not over.
 In other words, an instance may never violate the constraints of its parent.
-This property makes it very hard to inadvertendly make false conclusions in CUE.
+This property makes it very hard to inadvertently make false conclusions in CUE.
 Default values do not change this property; they syntactically appear as
 non-concrete values.
 CUE also bails out and requires explicit values if two conflicting defaults
 are specified for the same field, again limiting the search space.
+<!-- (jba) There's no actual searching going on in the implementation (IIUC), so using the term "search
+space" is confusing. -->
 
 With approaches that allow overrides, whether it be the complex inheritance
 used in languages like GCL and Jsonnet
 or the much simpler file-based approaches as used in HCL and Kustomize,
 finding a declaration for a concrete field value does not guarantee
-a final answer.
+a final answer,
+because another concrete value that occurs elsewhere can override it.
 When one needs to change a value of such a field,
-it can be time consuming and,
-especially in a pressured situation,
+it can be time-consuming and,
+especially when under pressure,
 very tempting to skip following complicated inheritance chains,
-double check a configuration file specifying overlay order,
+double-check a configuration file specifying overlay order,
 or look for a file that is lexically sorted after the one under consideration.
+<!-- (jba) I think it's worth making the more general point that order-independence
+makes life much simpler. Compare with functional vs. imperative languages? -->
 
 So there is a clear benefit to having fully expanded configurations
 over such override methods.
 CUE simulates that benefit by guaranteeing that any observed field value
 holds for the final result.
 If a user makes a false assumption that a concrete (default)
 value is the last in a chain,
+<!-- (jba) "last in a chain" makes it sound like order matters -->
```

Comment:
`If the user makes the false assumption that no concrete value is specified to discard the default value,`

perhaps?